driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -1319,6 +1327,10 @@ protected void initChannel(SocketChannel channel) throws Exception {
             pipeline.addLast("frameDecoder", new Frame.Decoder());
             pipeline.addLast("frameEncoder", frameEncoder);
 
+            if (useBeta) {
+                pipeline.addLast(new Frame.BetaVersionFlagAdder());
```

Comment:
Very minor suggestions:

1. I don't know the practical implications of _not_ giving a name to a handler, but since all the other handlers are given names in this method, I'd suggest we name this new one too.
2. Since this handlers is stateless, couldn't we make it a singleton?